lib/core/data/payment/payment_repository_impl.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:app/core/data/event/gql/event_tickets_query.dart';
+import 'package:app/core/data/payment/dtos/payment_card_dto/payment_card_dto.dart';
+import 'package:app/core/domain/payment/entities/payment_card_entity/payment_card_entity.dart';
+import 'package:app/core/domain/payment/payment_repository.dart';
+import 'package:app/core/failure.dart';
+import 'package:app/core/utils/gql/gql.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:dartz/dartz.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton(as: PaymentRepository)
+class PaymentRepositoryImpl extends PaymentRepository {
+  final _client = getIt<AppGQL>().client;
+
+  @override
+  Future<Either<Failure, PaymentCardEntity>> createNewCard({
+    required String tokenId,
+  }) async {
+    final result = await _client.mutate(
+      MutationOptions(
+        document: createNewCardQuery,
+        variables: {
+          'payment_method': tokenId,
+        },
+        parserFn: (data) => PaymentCardEntity.fromDto(
+          PaymentCardDto.fromJson(data['createStripeCard']),
+        ),
+      ),
+    );
+    if (result.hasException) {
+      return Left(Failure.withGqlException(result.exception));
+    }
+    return Right(result.parsedData!);
+  }
+
+  @override
+  Future<Either<Failure, dynamic>> createNewPayment() {
+    // TODO: implement createNewPayment
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<Either<Failure, List<PaymentCardEntity>>> getListCard() async {
+    final result = await _client.query(
+      QueryOptions(
+        document: getListCardQuery,
+        variables: const {
+          'skip': 0,
+          'limit': 20,
+          'provider': 'local',
```

Comment:
GetListCardInput ?

Reply:
For card payments, ideally, we should not store too many cards to avoid leaking too much sensitive information. I'm seeing most of e-commerce apps store up to 4 credit cards. 
Therefore, there will be no editing on those 3 variables, create input for it is unnecessary